### PR TITLE
Added H2 database and configured for testing.

### PR DESCRIPTION
**Pull request by vorfolomey**
#1438

### CODE REVIEW

File: core/src/main/resources/db/changelog/logs/ch-add-tipsandtricks-comment-Beshlei.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
     </changeSet>
 
     <changeSet id="Beshlei-2" author="Taras Beshlei">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <foreignKeyConstraintExists foreignKeyName="fk_comment_parent" />
```
Comment (antonbakalets) [inline core/src/main/resources/db/changelog/logs/ch-add-tipsandtricks-comment-Beshlei.xml:38]: Why this change and many others like this? Is there another script that is also adding this?

File: core/src/test/resources/application.properties [hunk 2]
```diff
@@ -11,10 +11,10 @@ spring.liquibase.password=${DATASOURCE_PASSWORD}
 spring.liquibase.enabled=true
 spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.xml
 # Hibernate
-spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.hibernate.ddl-auto=update
```
Comment (antonbakalets) [inline core/src/test/resources/application.properties:14]: I just can approve due to this change. Even if it is a developer profile.

In this project only Liquibase is responsible to update the database to certain state.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: core/src/main/resources/db/changelog/db.changelog-master.xml [hunk 3]
```diff
@@ -51,4 +51,5 @@
     <include file="db/changelog/logs/ch-fact-of-the-day-Lehkyi.xml"/>
     <include file="db/changelog/logs/ch-social-network-Lehkyi.xml"/>
     <include file="db/changelog/logs/ch_rating_statictics_Dovganyuk.xml"/>
+<!--    <include file="db/changelog/logs/ch-add-constraints-Lehkyi-1.xml"/>-->
```
[the remark this answers is not in the document]
Comment (vorfolomey) [reply core/src/main/resources/db/changelog/db.changelog-master.xml]: I removed it

File: core/src/main/resources/db/changelog/logs/ch-add-tipsandtricks-comment-Beshlei.xml (hunk 1 above)
Comment (vorfolomey) [reply core/src/main/resources/db/changelog/logs/ch-add-tipsandtricks-comment-Beshlei.xml]: H2 database doesn't allow same foreign key name

File: core/src/test/resources/application.properties (hunk 2 above)
Comment (vorfolomey) [reply core/src/test/resources/application.properties]: Fix issues with h2 real -> double

**Review by TarasDovganyuk — APPROVED** [review]

**Review by antonbakalets — APPROVED** [review]

**Review by eyouu — APPROVED** [review]
